Correcting type for encoding parameter in typescript/fasta

dungwinux opened the pull request:
Running old code will result in the following errors:

```
fasta.ts:24:52 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.

24             process.stdout.write(this.buf.toString(ENCODING, 0, this.ct));
                                                      ~~~~~~~~

fasta.ts:42:48 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.

42     out.buf.write(title, out.ct, title.length, ENCODING);
                                                  ~~~~~~~~

fasta.ts:67:48 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'BufferEncoding'.

67     out.buf.write(title, out.ct, title.length, ENCODING);
                                                  ~~~~~~~~
```

Because `BufferEncoding` is an union of string type, and `ENCODING` was defined as a string variable (`var ... ENCODING = 'binary'`), the `BufferEncoding` type was actually a subset of the type of `ENCODING`, thus the error.

This fix defines `ENCODING` as a constant, guaranteeing it to be within the `BufferEncoding` type.

While explicitly coercing as `BufferEncoding` is possible, I don't know where the type is defined to import.

nicovank reviewed (approved):
🤦 Looked through the current CLBG source, and it looks like their (only) TS version is not this one. We should make note that we changed the source for this benchmark (this PR is enough for now), is there any other benchmarks whose source we've altered from the original repo? I don't think so.

dungwinux commented:
> 🤦 Looked through the current CLBG source, and it looks like their (only) TS version is not this one. We should make note that we changed the source for this benchmark (this PR is enough for now), is there any other benchmarks whose source we've altered from the original repo? I don't think so.

To nitpick, we added a newline at the end in `Python/binary-trees/binarytrees.py` and remove `/// <reference path="../node_modules/@types/node/index.d.ts" />` in all TypeScript sources

nicovank commented:
Thanks for reminding me. The newline is whatever, we can restore that. We may end up making note of that other TypeScript change as well.
